src/WikiService.Data/RubricRepository.cs

```diff
@@ -112,5 +88,34 @@ public async Task<bool> DoesRubricNameExistAsync(Guid? rubricParentId, string na
     {
       return await _provider.Rubrics.AnyAsync(p => p.ParentId == rubricParentId && p.Name == nameRubric);
     }
+
+    private IQueryable<DbRubric> FindRubricChild(IQueryable<DbRubric> dbRubric)
+    {
+      foreach (DbRubric topRubric in dbRubric.ToList())
+      {
+        foreach (DbRubric rubric in _provider.Rubrics.AsEnumerable())
```

Comment:
Isn't possible to join FOREACH with IF and put FOREACH inside if-condition with using lambda-expression and Any()?
(I'm just asking; really don't know is it possible or not. But it can be interesting to try)

Reply:
The current solution is optimal in this case.